website/addons/googledrive/views/config.py

```diff
@@ -20,11 +20,17 @@
 @must_have_permission(permissions.WRITE)
 def googledrive_config_get(node_addon, auth, **kwargs):
     """API that returns the serialized node settings."""
-    result = GoogleDriveSerializer(
+    serializer = GoogleDriveSerializer(
         node_settings=node_addon,
         user_settings=auth.user.get_addon('googledrive')
-    ).serialized_node_settings
-    return result
+    )
+    if node_addon.external_account:
+        result = serializer.serialize_settings(node_addon, auth.user)
```

Comment:
Could `serialize_settings` handle the case where `node_addon.external_account` is `None`? Then you wouldn't need to use `serialized_node_settings` here.